### PR TITLE
Test for all hosts page manage columns

### PR DESCRIPTION
**Pull request by sambible**
### Problem Statement
Automation for 6.16 feature work as part of the All Hosts pt2 improvement.

### Related Issues
https://github.com/SatelliteQE/airgun/pull/1421 - airgun support PR

### PRT 
trigger: test-robottelo
pytest: tests/foreman/ui/test_host.py -k 'test_all_hosts_manage_columns'
airgun: 1421

### CODE REVIEW

**Comment by sambible** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_host.py -k 'test_all_hosts_manage_columns'

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7337
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/ui/test_host.py -k test_all_hosts_manage_columns --external-logging
Test Result : ========== 1 failed, 50 deselected, 60 warnings in 806.67s (0:13:26) ===========
```

**Comment by sambible** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_host.py -k 'test_all_hosts_manage_columns'
airgun: 1421

**Comment by sambible** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_host.py -k 'test_all_hosts_manage_columns'
airgun: 1421

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7340
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/ui/test_host.py -k test_all_hosts_manage_columns --external-logging
Test Result : ========== 1 passed, 50 deselected, 60 warnings in 835.34s (0:13:55) ===========
```

File: tests/foreman/ui/test_host.py [hunk 1]
```diff
@@ -1849,6 +1880,37 @@ def test_all_hosts_bulk_delete(target_sat, function_org, function_location, new_
         assert session.all_hosts.bulk_delete_all()
 
 
+@pytest.mark.tier2
+def test_multi_cv_read(target_sat, rhel_contenthost, function_org, new_host_ui):
```
Comment (pondrejk) [inline tests/foreman/ui/test_host.py:1884]: not sure if this was meant to be included, the description and prt run don't mention it

File: tests/foreman/ui/test_host.py [hunk 2]
```diff
@@ -1203,6 +1203,37 @@ def test_positive_manage_table_columns(
             assert (column in displayed_columns) is is_displayed
 
 
+@pytest.mark.tier2
+def test_all_hosts_manage_columns(target_sat, function_org, new_host_ui):
```
[the remark this answers is not in the document]
Comment (pondrejk) [reply tests/foreman/ui/test_host.py:1207]: or better, link from master https://github.com/pnovotny/robottelo/blob/0e6cce4df323abe5a9fcf4db88c2672cc6d9cd74/tests/foreman/ui/test_host.py#L1160

**Review by lhellebr — COMMENTED** [review]
Indeed, it seems like the first part of this PR is a duplicate and the second was added by mistake and should be part of another PR.

File: tests/foreman/ui/test_host.py (hunk 1 above)
Comment (sambible) [reply tests/foreman/ui/test_host.py:1884]: You are right, this is indeed cruft from some additional work. I'll clean this up .

File: tests/foreman/ui/test_host.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (sambible) [reply tests/foreman/ui/test_host.py:1207]: This is coverage for a similar feature, on a different page - the new All Hosts page. It's a rewrite of the original Hosts page, and until that page is retired, it needs to be covered as well.

**Comment by lhellebr** [comment]
Would it be better to have a single test, parametrized to use different Airgun entities, one for Hosts and one for All Hosts? Just a question, non-blocking.

**Review by lhellebr — APPROVED** [review]

File: tests/foreman/ui/test_host.py [hunk 3]
```diff
@@ -1203,6 +1203,37 @@ def test_positive_manage_table_columns(
...
+    """Verify that the manage columns widget changes the columns appropriately
+
+    :id: 5e13267a-68d2-451a-ae00-6502dd5db7f4
+
+    :expectedresults: Through the widget you can change the columns on the All Hosts page
+
+    :CaseComponent:Hosts-Content
+
+    :Team: Phoenix-subscriptions
+    """
```
Comment (vsedmik) [inline tests/foreman/ui/test_host.py:1217]: We could add the RFE-to-test mapping.
```suggestion
    
    :Verifies: SAT-19064
    """
```

File: tests/foreman/ui/test_host.py [hunk 4]
```diff
@@ -1203,6 +1203,37 @@ def test_positive_manage_table_columns(
...
 
+@pytest.mark.tier2
+def test_all_hosts_manage_columns(target_sat, function_org, new_host_ui):
+    """Verify that the manage columns widget changes the columns appropriately
+
+    :id: 5e13267a-68d2-451a-ae00-6502dd5db7f4
+
+    :expectedresults: Through the widget you can change the columns on the All Hosts page
+
+    :CaseComponent:Hosts-Content
```
Comment (vsedmik) [inline tests/foreman/ui/test_host.py:1214]:
```suggestion
    :CaseComponent: Hosts-Content
```

**Review by vsedmik — APPROVED** [review]
Looks good! Just a couple of nitpicks left.

File: tests/foreman/ui/test_host.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (pondrejk) [reply tests/foreman/ui/test_host.py:1207]: all right, fair enough

**Comment by sambible** [comment]
> Would it be better to have a single test, parametrized to use different Airgun entities, one for Hosts and one for All Hosts? Just a question, non-blocking.

This would be a more elegant solution, yes. The cases where the two tests look this similar though, are not that common. Seems worth investigating, though. (outside this PR)

**Pull request merged**